ReactWindows/ReactNative.Shared/Common/AccessibilityHelper.cs

```diff
@@ -0,0 +1,229 @@
+                    SetChildrenAccessibilityView(uiElementAutomationPeer, AccessibilityView.Raw);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Attached property used to store ImportantForAccessibility value in native controls.
+        /// </summary>
+        public static readonly DependencyProperty ImportantForAccessibilityAttachedProperty = DependencyProperty.RegisterAttached(
+                                                                                                "ImportantForAccessibilityAttached",
+                                                                                                typeof(ImportantForAccessibility),
+                                                                                                typeof(UIElement),
+                                                                                                new PropertyMetadata(ImportantForAccessibility.Auto));
+
+        /// <summary>
+        /// ImportantForAccessibilityAttached property setter.
+        /// </summary>
+        /// <param name="element"></param>
+        /// <param name="value"></param>
+        public static void SetImportantForAccessibilityAttached(UIElement element, ImportantForAccessibility value)
+        {
+            element.SetValue(ImportantForAccessibilityAttachedProperty, value);
+        }
+
+        /// <summary>
+        /// ImportantForAccessibilityAttached property getter.
+        /// </summary>
+        /// <param name="element"></param>
+        /// <returns></returns>
+        public static ImportantForAccessibility GetImportantForAccessibilityAttached(UIElement element)
+        {
+            return (ImportantForAccessibility)element.GetValue(ImportantForAccessibilityAttachedProperty);
+        }
+
+        /// <summary>
+        /// Recursively sets AccessibilityView to accessibilityView for parentAutomationPeer children.
+        /// </summary>
+        /// <param name="parentAutomationPeer"></param>
+        /// <param name="accessibilityView"></param>
+        private static void SetChildrenAccessibilityView(AutomationPeer parentAutomationPeer, AccessibilityView accessibilityView)
+        {
+            if (parentAutomationPeer?.GetChildren() == null)
+            {
+                return;
+            }
+
+            foreach (AutomationPeer childAutomationPeer in parentAutomationPeer.GetChildren())
+            {
+                if (childAutomationPeer is FrameworkElementAutomationPeer childFEAutomationPeer)
+                {
+                    if (childFEAutomationPeer?.Owner != null)
+                    {
+                        AutomationProperties.SetAccessibilityView(childFEAutomationPeer.Owner, accessibilityView);
+                    }
+                }
+                else if (childAutomationPeer is ItemAutomationPeer childIAutomationPeer)
```

Comment:
>ItemAutomationPeer [](start = 48, length = 18)

looks like the pattern of recursing through children while doing different actions for each child depending on if it has ItemAutomationPeer or FrameworkAutomationPeer is repeated three times in this class. Maybe it's possible to refactor this logic into a helper method which will take two Action's as what to do for frameworkElement or item peers?

Reply:
I think there's more in common with them than that. It looks like they all are trying to convert an `AutomationPeer` to a `UIElement` so perhaps we should move the logic into a helper called `GetElementFromAutomationPeer`.

---
In reply to: [163125697](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r163125697) [](ancestors = 163125697)

Reply:
Moved the common logic into a separate method

---
In reply to: [163336908](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r163336908) [](ancestors = 163336908,163125697)